src/keyfile.c

```diff
@@ -1313,11 +1316,47 @@ void configuration_apply_settings(void)
 }
 
 
+static gboolean save_configuration_cb(gpointer data)
+{
+	configuration_save();
```

Comment:
Just continuing @b4n's theme, do we need to save both config and project?  The active session is in the project, not the config.

Reply:
And on that note, shouldn't the config be saved on project open/close as well, although it will often happen anyway because files will be opened/closed when a project opens, but not always.

Reply:
> Just continuing @b4n's theme, do we need to save both config and project? The active session is in the project, not the config.

Maybe we do not need to save the config if a project is open regarding the session files but for all other maybe changed settings and I think it won't hurt(except disk IO) to leave a consistent state on disk.

> And on that note, shouldn't the config be saved on project open/close as well, although it will often happen anyway because files will be opened/closed when a project opens, but not always.

In my tests, the `save_configuration_cb()` was always called on project related actions. What exact cases do you refer to by "not always"?

Reply:
> "not always"

On further consideration since any project action closes one set of files and opens another (even if they are the same files) there will be file open/close signals in all cases except the corner case where no files are open before and after the project open/close, so don't worry about it I think.